de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/fskdbview/FSKDBViewNodeModel.java

```diff
@@ -231,14 +241,19 @@ public void saveCurrentValue(NodeSettingsWO content) {
       throws Exception {
     PortObject inPort = inObjects[0];
     PortObject outputPort = null;
+
     synchronized (getLock()) {
       FSKDBViewRepresentation representation = getViewRepresentation();
       representation.setRemoteRepositoryURL(m_repositoryLocationSettings.getStringValue());
       representation.setMaxSelectionNumber(
           ((SettingsModelIntegerBounded) m_maxSelectionNumberSettings).getIntValue());
-      if (inPort == null) {
+
+      if (inPort == null && representation.getTable() == null) {
         // if the optional input port is not provided then
         outputPort = createEmptyTable(exec);
+        JSONDataTable jsonTable =
+            JSONDataTable.newBuilder().setDataTable((DataTable) outputPort).build(exec);
```

Comment:
What is m_table in FSKDBViewRepresentation? I see that you assign it the empty table created in performExecute but what do you use it for?

Reply:
This is important to have the base structure of BufferedDataTable without data and later will be filled with the data received from the online repository in the case of online repository fetch.